File: 240417/괄호 쌍 만들어주기 2/pair-parentheses-2.py
a_str = input()
n = len(a_str)

cnts, result = 0, 0
for i in range(n - 1, 0, -1):
    if a_str[i] == ')' and a_str[i - 1] == ')':
        cnts += 1
    elif a_str[i] == '(' and a_str[i - 1] == '(':
        result += cnts

print(result)

File: 260211/괄호 쌍 만들어주기 2/pair-parentheses-2.cpp
#include <iostream>
#include <string>
using namespace std;

int main() 
{
    string A;
    cin >> A;

    // Please write your code here.
    int N = A.length();
    int cnts = 0, result = 0;
    for(int i = N - 1; i > 0; i--)
    {
        if(A[i] == ')' && A[i - 1] == ')')
            cnts += 1;
        else if(A[i] == '(' and A[i - 1] == '(')
            result += cnts;
    }

    cout << result;
    return 0;
}